src/vos/vos_io.c

```diff
@@ -491,8 +492,12 @@ akey_fetch_recx(daos_handle_t toh, const daos_epoch_range_t *epr,
 			if (rc != 0)
 				return rc;
 			biov_align_lens(&biov, ent, rsize);
+			csum_enabled = true;
 		} else {
 			bio_iov_set_extra(&biov, 0, 0);
+			if (csum_enabled)
+				D_ERROR("Checksum found in some entries, "
```

Comment:
Any reason not to assert this?   Is there any non-bug reason this would happen?   The issue with such errors, is they will likely not be noticed.

Reply:
Guess I didn't want to assert the server. It would still end up returning a CSUM.